github/resource_github_team_repository.go

```diff
@@ -128,7 +129,11 @@ func resourceGithubTeamRepositoryRead(d *schema.ResourceData, meta interface{})
 	}
 
 	d.Set("etag", resp.Header.Get("ETag"))
-	d.Set("team_id", teamIdString)
+	if d.Get("team_id") == "" {
+		// If team_id is empty, that means we are importing the resource.
+		// Set the team_id to be the id of the team.
+		d.Set("team_id", teamIdString)
+	}
```

Comment:
This seems like the riskiest bit.  What was your experience leading up to this diff?

Reply:
In my testing, I didn't have this if statement at first.

I tried setting the `team_id` to be a team slug, and was able to plan and apply the following config to create a new team repo connection:
```
resource "github_team_repository" "test" {
    team_id = "sampleteam"
    repository = "testteamrepo"
    permission = "push"
}
```

When I tried another plan after that I expected that terraform would say that there would be no changes necessary. However, I got the following output:
```
Terraform will perform the following actions:

  # github_team_repository.test must be replaced
-/+ resource "github_team_repository" "test" {
      ~ etag       = "W/\"6469d39bbad910892556bd4bb130c23ab7ffb655ec6a312d592f2e76736d67c9\"" -> (known after apply)
      ~ id         = "4466024:testteamrepo" -> (known after apply)
        permission = "push"
        repository = "testteamrepo"
      ~ team_id    = "4466024" -> "123" # forces replacement
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

When terraform refreshed the state, using this `resourceGithubTeamRepositoryRead` function, it would replace the `team_id` from a slug to the actual team id, making it think that it would need to replace this resource.

So I added this if statement to avoid this situation. The only situation I could think of where we would want to actually set this `team_id` field here, would be when we were importing a team_repo resource.

Reply:
Appreciate the detail here, thanks.